docs/release/release-tasks.md

```diff
@@ -0,0 +1,384 @@
+       docker pull registry.k8s.io/cluster-api/kubeadm-bootstrap-controller:${RELEASE_TAG}
+       docker pull registry.k8s.io/cluster-api/kubeadm-control-plane-controller:${RELEASE_TAG}
+       ```
+3. Publish the release in GitHub:
+    1. Get the final release notes from the docs team and add them to the GitHub release.
+    2. Publish the release (ensure to flag the release as pre-release if necessary).
+4. Publish `clusterctl` to Homebrew by bumping the version in [clusterctl.rb](https://github.com/Homebrew/homebrew-core/blob/master/Formula/clusterctl.rb).
+   **Notes**:
+    * This is only done for new latest stable releases, not for beta / RC releases and not for previous release branches.
+    * For an example please see: [PR: clusterctl 1.1.5](https://github.com/Homebrew/homebrew-core/pull/105075/files).
+    * Homebrew has [conventions for commit messages](https://docs.brew.sh/Formula-Cookbook#commit) usually
+      the commit message for us should look like: `clusterctl 1.1.5`.
+5. Set EOL date for previous release (prior art: https://github.com/kubernetes-sigs/cluster-api/issues/7146).
+
+Additional information:
+* [Versioning documentation](./../../CONTRIBUTING.md#versioning) for more information.
+* Cutting a release as of today requires permissions to:
+    * Create a release tag on the GitHub repository.
+    * Create/update/publish GitHub releases.
+
+#### [Optional] [Track] Bump the ClusterAPI apiVersion
+
+**Note** This should only be done when we have to bump the apiVersion of our APIs.
+
+1. Add new version of the types:
+    1. Create new api packages by copying existing packages.
+    2. Make sure webhooks only exist in the latest apiVersion (same for other subpackages like `index`).
+    3. Add conversion and conversion tests.
+    4. Adjust generate targets in the Makefile.
+    5. Consider dropping fields deprecated in the previous apiVersion.
+2. Update import aliases in `.golangci.yml`.
+3. Switch other code over to the new version (imports across the code base, e.g. controllers).
+    1. Add all versions to the schema in the `main.go` files.
+4. Add types to the `PROJECT` files of the respective provider.
+5. Add test data for the new version in `test/e2e/data/{infrastructure-docker,shared}` (also update top-level `.gitignore`).
+6. Update `docker.yaml`, make sure all tests are successful in CI.
+
+#### [Optional] [Track] Bump the Kubernetes version
+
+TODO(sbueringer): Follow-up PR (this will probably just be create an issue based on another issue template)
+
+## Communications/Docs/Release Notes Manager
+
+### Responsibilities
+
+* Communication:
+    * Communicate key dates to the community
+* Documentation:
+    * Improve release process documentation
+    * Ensure the book and provider upgrade documentation are up-to-date
+    * Maintain and improve user facing documentation about releases, release policy and release calendar
+* Release Notes:
+    * Polish release notes
+
+### Tasks
```

Comment:
Can we also add a doc to keep this and other release process docs updated if anything changes?

Reply:
I can add a point to the issue template under Continuously and I can add it here as well. 

But not sure about the details. Essentially I can only write keep the documents up-to-date if anything changes

Reply:
I added the following to Continuously in the issue template:
* [Communications Manager] Improve release process documentation
* [Communications Manager] Maintain and improve user facing documentation about releases, release policy and release calendar

I would only add a corresponding section if we have additional details to document